### PR TITLE
relocate floater when params change, layout window changes

### PR DESCRIPTION
**Pull request by mouEsam**

### CODE REVIEW

File: lib/src/easy_draggable_widget.dart [hunk 1]
```diff
@@ -124,25 +128,32 @@ class _EasyDraggableWidgetState extends State<EasyDraggableWidget>
           child: LayoutBuilder(
             builder: (context, constraints) {
               final size = constraints.biggest;
+              windowSize = size;
               final height = size.height;
               final width = size.width;
 
               double getTop() {
-                return this.top ??
+                var top = this.top ??
```
Comment (YousefSalahKassem) [inline lib/src/easy_draggable_widget.dart:136]: why you need to use var ?

File: lib/src/easy_draggable_widget.dart [hunk 2]
```diff
@@ -124,25 +128,32 @@ class _EasyDraggableWidgetState extends State<EasyDraggableWidget>
...
+                    });
+                if (widget.constrainToWindow && top != null) {
+                  top = top.clamp(0, height - (this.floatingSize?.height ?? 0)) as double;
+                }
+                return top ?? -1;
               }
 
               double getLeft() {
-                return this.left ??
+                var left = this.left ??
```
Comment (YousefSalahKassem) [inline lib/src/easy_draggable_widget.dart:148]: same as here

**Review by YousefSalahKassem — CHANGES_REQUESTED** [review]
Almost LGTM as I learned too much from you.
but please check my comments

File: lib/src/easy_draggable_widget.dart (hunk 1 above)
Comment (mouEsam) [reply lib/src/easy_draggable_widget.dart:136]: On the next line it gets clamped, ie. changed.

**Comment by YousefSalahKassem** [comment]
LGTM!

**Pull request merged**